### PR TITLE
[constexpr] alg.copy

### PR DESCRIPTION
**Pull request by gnzlbg**

### CODE REVIEW

**Comment by gnzlbg** [comment]
Depends on #725 .

File: test/algorithm/copy.cpp [hunk 1]
```diff
@@ -67,5 +77,9 @@ int main()
         CHECK(std::strcmp(sz, buf) == 0);
     }
 
+#if RANGES_CXX_CONSTEXPR >= RANGES_CXX_CONSTEXPR_14
```
Comment (CaseyCarter) [inline test/algorithm/copy.cpp]: I would replace line 21 with this line...

File: test/algorithm/copy.cpp [hunk 2]
```diff
@@ -67,5 +77,9 @@ int main()
...
+    static_assert(test_constexpr_copy(), "");
+#endif
```
Comment (CaseyCarter) [inline test/algorithm/copy.cpp]: ...and then move these two lines to line 29. (i.e., keep the `constexpr` test together as a single unit.)

**Comment by gnzlbg** [comment]
@CaseyCarter I feel a bit bad about duplicating tests for constexpr purposes.

File: test/algorithm/copy.cpp (hunk 2 above)
Comment (gnzlbg) [reply test/algorithm/copy.cpp]: done

**Comment by gnzlbg** [comment]
@ericniebler FWIW it worked (all green), so IIUC what travis does is apply the commits in the pull-request to the receiving branch (in this case `ericniebler/range-v3:master`).

This is both good and bad. Good because I can send a PR, branch it, keep on working on something that needs those changes, and when that is done, cherry-pick into a clean branch, and pull request that. Test will initially fail, but once the dependent branches are merged they will pass.

Bad because a PR becomes "green" when the tests pass. But if you merge a PR, the tests on all other green PRs won't be re-run, so merging subsequent pull-requests can still break the build. This leads to "absurd" situations in which one PR is green but has merge conflicts (how can the tests pass if the PR can't even be merged?).

**Pull request merged**